From ba1be15b0850aacb16d78949ef6a2efb11c1fe87 Mon Sep 17 00:00:00 2001
From: Simeon Marijon <simeon.marijon@st.com>
Date: Fri, 21 Feb 2025 17:23:53 +0100
Subject: [PATCH 1564/1695] module/scmi: assert that device->element_id is a
 valid_element_id

Previously the function called test but return no error

outligned by coverity

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git fc1445e8de69a60ab809b420805f768e05a2c47f module/scmi: assert that device->element_id is a valid_element_id]
Signed-off-by: Simeon Marijon <simeon.marijon@st.com>
Change-Id: Ifda02fa39cbb76aea520a6c8111f519712616d4d
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/SCP-firmware/+/408742
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 .../module/scmi_voltage_domain/src/mod_scmi_voltage_domain.c   | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/core/lib/scmi-server/SCP-firmware/module/scmi_voltage_domain/src/mod_scmi_voltage_domain.c b/core/lib/scmi-server/SCP-firmware/module/scmi_voltage_domain/src/mod_scmi_voltage_domain.c
index f38d35040..a299d1b17 100644
--- a/core/lib/scmi-server/SCP-firmware/module/scmi_voltage_domain/src/mod_scmi_voltage_domain.c
+++ b/core/lib/scmi-server/SCP-firmware/module/scmi_voltage_domain/src/mod_scmi_voltage_domain.c
@@ -158,7 +158,8 @@ static int get_device(fwk_id_t service_id, unsigned int elt_idx,
         return FWK_E_RANGE;
 
     device = &agent->device_table[elt_idx];
-    fwk_module_is_valid_element_id(device->element_id);
+
+    fwk_assert(fwk_module_is_valid_element_id(device->element_id));
 
     if (out_device)
         *out_device = device;
-- 
2.47.2

